src/modules/nfts/components/wizard/MintInProgress.tsx

```diff
@@ -0,0 +1,334 @@
+        },
+        onComplete: () => {
+          console.log('signing complete');
+          setTransactionStep(TransactionStep.SUCCESS);
+          handleNext();
+        },
+        onError: (msg) => {
+          throw new Error(msg);
+        },
+      });
+    } catch (err) {
+      console.log('signing err', err);
+      setTransactionStep(TransactionStep.SIGNING_FAILED);
+    }
+  }, [setTransactionStep, mintResp, handleNext]);
+
+  const attemptMint = useCallback(
+    async (metaData: UploadedFilePin) => {
+      if (!metaData) {
+        throw new Error('No Meta Data, something went wrong');
+      }
+
+      setTransactionStep(TransactionStep.APPROVING);
+
+      try {
+        const mintResp = await mintNFT({
+          connection,
+          wallet,
+          uri: metaData.uri,
+          maxSupply: nftValue.properties.maxSupply,
+        });
+        setMintResp(mintResp);
+        setTransactionStep(TransactionStep.FINALIZING);
+        await connection.confirmTransaction(mintResp.txId);
+        setTransactionStep(TransactionStep.SUCCESS);
+      } catch (err) {
+        if (err?.code === APPROVAL_FAILED_CODE) {
+          setTransactionStep(TransactionStep.APPROVAL_FAILED);
+        } else {
+          setTransactionStep(TransactionStep.SENDING_FAILED);
+        }
+
+        return;
+      }
+
+      setTransactionStep(TransactionStep.SIGNING);
+    },
+    [nftValue, connection, wallet]
+  );
+
+  const attemptMetaDataUpload = useCallback(async () => {
+    if (!nftValue) {
+      throw new Error('No NFT Value, something went wrong');
+    }
+
+    if (nftValue.mintStatus === MintStatus.SUCCESS) {
+      return; // Don't accidentally mint
+    }
+
+    let metaData: UploadedFilePin | null = null;
+
+    try {
+      metaData = await uploadMetaData(nftValue);
+
+      attemptMint(metaData);
+    } catch (err) {
+      setTransactionStep(TransactionStep.META_DATA_UPLOAD_FAILED);
+    }
+  }, [attemptMint, nftValue, uploadMetaData, setTransactionStep]);
+
+  useEffect(() => {
+    if (showErrors) {
+      return;
+    }
+
+    if (isActive && nftValue) {
+      if (transactionStep === TransactionStep.META_DATA_UPLOADING) {
+        attemptMetaDataUpload();
+      } else if (transactionStep === TransactionStep.SIGNING) {
+        attemptHolaplexSign();
+      }
+    }
+  }, [
+    isActive,
+    showErrors,
+    nftValue,
+    transactionStep,
+    attemptHolaplexSign,
+    attemptMetaDataUpload,
+    index,
+  ]);
+
+  if (!nftValue) {
+    return null;
+  }
+
+  return (
+    <NavContainer
+      title={`Minting ${index + 1} of ${images.length}`}
+      previousStep={previousStep}
+      goToStep={goToStep}
+    >
+      <Row>
+        <Col style={{ marginRight: 224 }}>
+          <Row>
+            <Paragraph style={{ fontWeight: 900 }}>Progress</Paragraph>
+          </Row>
+          <Row style={{ marginTop: 37 }}>
+            <Col style={{ minWidth: 237 }}>
+              <MintStep
```

Comment:
antd has a steps component. Something we can refactor to later. Help cut down on boiler plate component code.

https://ant.design/components/steps/